src/rpc/handlers/LedgerEntry.cpp

```diff
@@ -37,7 +37,15 @@ doLedgerEntry(Context const& context)
         if (!key.parseHex(request.at(JS(index)).as_string().c_str()))
             return Status{Error::rpcINVALID_PARAMS, "malformedIndex"};
     }
-    else if (request.contains(JS(account_root)))
+    if (request.contains(JS(ledger_index)))
+    {
+        if (!request.at(JS(ledger_index)).is_string())
```

Comment:
Seems like the logic is incorrect here. Now if someone passes a `ledger_index` but it's not a string it will straight return an error, not even trying to check whether it's an integer or not. You have to check for both and return an error only if index is neither string nor integer.